docs/reference/pip_install.rst

```diff
@@ -337,45 +337,28 @@ the project path.  This is one advantage over just using ``setup.py develop``,
 which creates the "egg-info" directly relative the current working directory.
 
 
+Build System Interface
+++++++++++++++++++++++
 
-Controlling setup_requires
-++++++++++++++++++++++++++
-
-Setuptools offers the ``setup_requires`` `setup() keyword
-<http://pythonhosted.org/setuptools/setuptools.html#new-and-changed-setup-keywords>`_
-for specifying dependencies that need to be present in order for the `setup.py`
-script to run.  Internally, Setuptools uses ``easy_install`` to fulfill these
-dependencies.
-
-pip has no way to control how these dependencies are located.  None of the
-Package Index Options have an effect.
-
-The solution is to configure a "system" or "personal" `Distutils configuration
-file
-<http://docs.python.org/2/install/index.html#distutils-configuration-files>`_ to
-manage the fulfillment.
-
-For example, to have the dependency located at an alternate index, add this:
-
-::
+In order for pip to install a package from source, pip must recognise the build
+system. Today only one build system is recognised, with two variants.
 
-  [easy_install]
-  index_url = https://my.index-mirror.com
+If there is a ``setup.cfg`` with any of ``[extras]``, ``requires-dist`` or
+``requires-setup`` present then a declarative setuptools package is detected.
 
-To have the dependency located from a local directory and not crawl PyPI, add this:
+Otherwise there is a ``setup.py`` then non-declarative setuptools is assumed.
 
-::
+Declarative setuptools
+~~~~~~~~~~~~~~~~~~~~~~
 
-  [easy_install]
-  allow_hosts = ''
-  find_links = file:///path/to/local/archives
+``setup.py`` must implement the following commands::
 
+    setup.py install --record XXX [--single-version-externally-managed] [--root XXX] [--compile|--no-compile] [--install-headers XXX]
 
```

Comment:
I know you just described setup.cfg above - but perhaps a line stating that a setup.cfg with the keys must be present so that the section is self-complete?